src/pages/product-master/Forms/productLegForm.js

```diff
@@ -66,71 +87,105 @@ const ProductLegForm = ({ store, labels, expanded, editMode, maxAccess }) => {
-            toAmount: yup.string().required('to Amount is required'),
-            fromAmount: yup.string().required('From Amount is required')
+            fromAmount: yup.string().required(),
+            toAmount: yup.string().required()
           })
         )
-        .required('productLegs array is required')
+        .required(' ')
     }),
     onSubmit: values => {
       post(values.productLegs)
     }
   })
 
-  const columns = [
-    {
-      component: 'numberfield',
-      label: labels.fromAmount,
-      name: 'fromAmount'
-    },
-    {
+  const getColumns = async () => {
+    const response = await getRequest({
+      extension: CurrencyTradingSettingsRepository.CommissionType.qry,
+      parameters: '_filter='
+    })
+
+    const result = await response.list.map(item => ({
       component: 'numberfield',
-      label: labels.toAmount,
-      name: 'toAmount'
-    },
-    {
-      component: 'button',
-      name: 'saved',
-      label: labels.commission,
-      onClick: (e, row) => {
-        stack({
-          Component: ProductLegCommissionForm,
-          props: {
-            labels: labels,
-            maxAccess: maxAccess,
-            row,
-            store
-          },
-          width: 600,
-
-          title: labels?.commission
-        })
+      label: item.name,
+      name: item.recordId.toString()
+    }))
+
+    const columns = [
+      {
+        component: 'numberfield',
+        label: labels.fromAmount,
+        name: 'fromAmount'
+      },
+      {
+        component: 'numberfield',
+        label: labels.toAmount,
+        name: 'toAmount'
       }
-    }
-  ]
+    ]
+
+    setCommission(result)
+
+    setCommissionColumns([...columns, ...result])
+  }
+
+  useEffect(() => {
+    getColumns()
+  }, [pId])
+
   useEffect(() => {
     _seqNo && getScheduleRange(_seqNo)
   }, [_seqNo])
 
-  const getScheduleRange = () => {
+  const getScheduleRange = async () => {
     const defaultParams = `_productId=${pId}&_seqNo=${_seqNo}`
-    var parameters = defaultParams
-    getRequest({
-      extension: RemittanceSettingsRepository.ProductScheduleRanges.qry,
-      parameters: parameters
-    })
-      .then(res => {
-        formik.setValues({ productLegs: [] })
-        if (res.list.length > 0)
-          formik.setValues({
-            productLegs: res.list?.map(({ ...rest }, index) => ({
-              id: index + 1,
-              saved: true,
-              ...rest
-            }))
+    const parameters = defaultParams
+
+    try {
+      const allCommissionFees = await getRequest({
+        extension: RemittanceSettingsRepository.ProductScheduleFees.qry, //qryPSF
+        parameters: `_productId=${pId}&_seqNo=${_seqNo}&_rangeSeqNo=${0}`
+      })
+
+      const res = await getRequest({
+        extension: RemittanceSettingsRepository.ProductScheduleRanges.qry,
+        parameters: parameters
+      })
+
+      const productLegsPromises = res.list.map(async (item, index) => {
```

Comment:
add it as res?.list?.map

Reply:
why?

Reply:
response might be null or undefined to prevent errors if getRequest returns null or undefined